cli/src/commands/extensions/api.rs

```diff
@@ -0,0 +1,256 @@
+use std::cell::RefCell;
+use std::ops::Deref;
+use std::path::Path;
+use std::pin::Pin;
+use std::rc::Rc;
+use std::str::FromStr;
+
+use anyhow::{anyhow, Context, Error, Result};
+use deno_core::{op, OpDecl, OpState};
+use futures::future::BoxFuture;
+
+use phylum_types::types::auth::{AccessToken, RefreshToken};
+use phylum_types::types::common::{JobId, ProjectId};
+use phylum_types::types::job::JobStatusResponse;
+use phylum_types::types::package::{
+    Package, PackageDescriptor, PackageStatusExtended, PackageType,
+};
+use phylum_types::types::project::ProjectDetailsResponse;
+
+use crate::commands::parse::{get_packages_from_lockfile, LOCKFILE_PARSERS};
+use crate::config::get_current_project;
+use crate::{api::PhylumApi, auth::UserInfo};
+
+/// Holds either an unawaited, boxed `Future`, or the result of awaiting the future.
+pub enum OnceFuture<T: Unpin> {
+    Future(BoxFuture<'static, T>),
+    Awaited(T),
+}
```

Comment:
When I asked if this needs to be `pub(crate)` I was asking if this needs to be `pub` at all. Based on my tests, this does not need to be public.